packages/default/scss/pivotgrid/_layout.scss

```diff
@@ -84,6 +83,10 @@
         border-left-style: solid;
     }
 
+    .k-pivot-table .k-grid-header-wrap {
+        margin-right: -1px;
```

Comment:
Why do we need this negative margin?

Reply:
The above line fixes this issue:
https://www.screencast.com/t/jjyyhnj00hCD

If there is a better approach to achieve the desired behavior, please, let me know :)